### PR TITLE
Throw an error when adding generated columns via an alter table

### LINKED ISSUE #3882: ALTER TABLE ... ADD COLUMN ignores AS clause
Since generated columns aren't supported yet, `alter table` should return an error if a user tries to add a generated column.

```
turso> create table t(a);
turso> alter table t add column b as (null);
turso> select * from sqlite_schema;
┌───────┬──────┬──────────┬──────────┬───────────────────────┐
│ type  │ name │ tbl_name │ rootpage │ sql                   │
├───────┼──────┼──────────┼──────────┼───────────────────────┤
│ table │ t    │ t        │        2 │ CREATE TABLE t (a, b) │
└───────┴──────┴──────────┴──────────┴───────────────────────┘
```

### PR DESCRIPTION
**Pull request by RS2007**
Fixes #3882 and #3887

```bash
turso> create table t(a);
turso> alter table t add column b as (null);
  × Parse error: Alter table does not support adding generated columns
```

```bash
turso> create table t(a); create table s(a);  alter table s add column b references t(a);
turso> select * from sqlite_master;
┌───────┬──────┬──────────┬──────────┬────────────────────────────────────────────────────────┐
│ type  │ name │ tbl_name │ rootpage │ sql                                                    │
├───────┼──────┼──────────┼──────────┼────────────────────────────────────────────────────────┤
│ table │ t    │ t        │        2 │ CREATE TABLE t (a)                                     │
├───────┼──────┼──────────┼──────────┼────────────────────────────────────────────────────────┤
│ table │ s    │ s        │        3 │ CREATE TABLE s (a, b, FOREIGN KEY (b) REFERENCES t(a)) │
└───────┴──────┴──────────┴──────────┴────────────────────────────────────────────────────────┘
turso> 
```

### CODE REVIEW

File: testing/alter_column.test [hunk 1]
```diff
@@ -238,3 +238,21 @@ do_execsql_test_on_specific_db {:memory:} alter-table-rename-pk-and-fk {
...
+# Adding a generated column via ALTER TABLE should error
+do_execsql_test_in_memory_error_content alter-table-add-generated-column-error {
+    CREATE TABLE t(a);
+    ALTER TABLE t ADD COLUMN b AS (NULL);
+} {
+    "Parse error: Alter table does not support adding generated columns"
+}
+
+# Add column with a foreign key reference and verify schema SQL
+do_execsql_test_on_specific_db {:memory:} alter-table-add-column-with-fk-updates-schema {
```
Comment (PThorpe92) [inline testing/alter_column.test:311]: Hey can we just add a couple more tests here for edge cases? 

like self-referencing and composite foreign keys?

Comment (RS2007) [reply testing/alter_column.test:311]: Yup added these two in the latest commit

**Review by PThorpe92 — APPROVED** [review]
